tcmur_cmd_handler.c

```diff
@@ -142,6 +142,10 @@ static int check_lba_and_length(struct tcmu_device *dev,
 	ret = check_iovec_length(dev, cmd, sectors);
 	if (ret)
 		return ret;
+    
+    if (cdb[0] == COMPARE_AND_WRITE) {
+        sectors = sectors/2;
+    }
```

Comment:
I think what you wanted to fix here was fixed already.

We want to be checking this from:

tcmur_handle_caw -> handle_caw_check (here sectors is whatever is in the cdb) -> check_lbas

so check_lba_and_length is never called for a CAW and we should not have to do the sectors divided by 2 trick.

Reply:
I have fixed it. Thanks for your review.

Reply:
OK

Reply:
> OK

Hey, I saw this reply, but was not sure what it is for.

Did you see my other comments with the patches I attached:

https://github.com/open-iscsi/tcmu-runner/pull/365#discussion_r172960667
https://github.com/open-iscsi/tcmu-runner/pull/365#discussion_r172963556

?

In those comments, I was asking that you drop some of your code, incorporate my code into your patchset, confirm it works for your tests and then repush. This way you still get the GitHub pull request credit.

If you do not have time, we can do the reverse where I take your patches, merge them with mine, add your signed-off, and then make a new pull request. I thought you might not want to do this because you would just get a signed off and it might not show up in your GitHub pull request stats.